demo/api.html

```diff
@@ -1,11 +1,212 @@
 <div class="container bs-docs-container">
   <div class="row">
     <div class="col-md-9" role="main">
+
       <div class="bs-docs-section">
-        <h1 id="basics" class="page-header">
-          Events
+        <h1 id="routers" class="page-header">
+          Routers
+          <a href="routers" class="anchorjs-link"><span class="anchorjs-icon"></span></a>
+        </h1>
+      </div>
+
+      <div class="bs-docs-section">
+        <h1 id="rules" class="page-header">
+          Rules
+          <a href="rules" class="anchorjs-link"><span class="anchorjs-icon"></span></a>
         </h1>
         <p>
+          Rules are objects that contain a regular expression to match the URL, it might be directly related to the
+          router object or being child of another rule.
+        </p>
+        <p>
+          When a URL change or is manually triggered, the rules are executed in order to check all changes that will
+          occur on the change of the current URL. Those changes create a queue of events to be executed that respect
+          the event priority order.
+        </p>
+        <p>
+          A <code>rule</code> instance is also a <code>HTMLElement</code>, witch means that it will have the same
+          structure, methods and properties of a <code>HTMLElement</code>, using browser native event listeners,
+          allowing be appended into other elements or have elements inside it.
+        </p>
+        <pre class="prettyprint lang-html"><xmp><pushstatetree-rule id="myRuleID" rule="/regular_expression/">
+  <pushstatetree-rule id="myParentRule" rule="/regular_expression/" parentGroup="1">
```

Comment:
it's because the `xmp` element there, if I ident correctly this part, it will break in the HTML since it will apply the spacing in the result rendering.